the-collab-lab/tcl-73-smart-shopping-list
# Create validation message that prompts a user to sign in if they want to use the app.

## Description

This code adds a prompt that pops up for the user whenever they try to perform an action(e.g. create a list) when not signed into the app.

Also.. I know I was not tasked with styling buttons, but some color styling for the prompt buttons, as well as the `create a list` and `submit` buttons was added, by experimenting with the method I outlined in the issue doc (also outlined [here](https://www.frontend.fyi/v/fixing-radix-themes-with-css-cascade-layers)). This can be removed before merging if needed.

## Related Issue

Closes #32 

## Acceptance Criteria
- [x] The app should display a prompt on each page that instructs the user to sign in if they want to view that page (e.g. "To create a list/view a list/manage a list, please sign in.").
  - Alternatively, consider automatically redirecting the user to a login page when they try to access a page they are not authorized to view.
- [x] When a user is signed in, they should be able to view all pages and interact with the app's features (creating lists, viewing lists, adding new items to lists).

## Type of Changes

Enhancement

## Updates

### Before
**User not signed in. No prompt when clicking on create list.**
![image](https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/78065152/acb1b04b-3590-48d9-9652-ddb5ae7d6c2a)

### After
**User not signed in. Sign in prompt appears when user attempts to create a list.**
![image](https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/78065152/262b6fb8-7c38-41c8-a587-9a4e5a384c5c)

## Testing Steps / QA Criteria

`git pull`
`git checkout ja-feature-signin-prompt`
`npm ci`
`npm start`

- If signed into the app, sign out.
- On the home page, try to create a list. Upon clicking `create a list` there should be a prompt for you to sign in.
  - Clicking `Sign In` should redirect you to your Google sign in page, where you can log in successfully.
- Navigate to the `Manage List` page and try to add an item. Upon clicking `Submit` there should be a prompt for you to sign in.
